packages/data-stores/src/site/test/selectors.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * These tests shouldn't require the jsdom environment,
+ * but we're waiting for this PR to merge:
+ * https://github.com/WordPress/gutenberg/pull/20486
+ *
+ * @jest-environment jsdom
+ */
+
+/**
+ * External dependencies
+ */
+import { dispatch, select, subscribe } from '@wordpress/data';
+import wpcomRequest from 'wpcom-proxy-request';
+
+/**
+ * Internal dependencies
+ */
+import { register } from '..';
+import { STORE_KEY } from '../constants';
+
+jest.mock( 'wpcom-proxy-request', () => ( {
+	__esModule: true,
+	default: jest.fn(),
+	requestAllBlogsAccess: jest.fn( () => Promise.resolve() ),
+} ) );
+
+let store: ReturnType< typeof register >;
+
+beforeAll( () => {
+	store = register( { client_id: '', client_secret: '' } );
+} );
+
+beforeEach( () => {
+	( wpcomRequest as jest.Mock ).mockReset();
+	dispatch( store ).reset();
+} );
+
+describe( 'getSite', () => {
+	it( 'resolves the state via an API call and caches the resolver on success', async () => {
+		const slug = 'mytestsite12345.wordpress.com';
+		const apiResponse = {
+			ID: 1,
+			name: 'My test site',
+			description: '',
+			URL: 'http://mytestsite12345.wordpress.com',
+		};
+
+		( wpcomRequest as jest.Mock ).mockResolvedValue( apiResponse );
+
+		const listenForStateUpdate = () => {
+			return new Promise( resolve => {
+				const unsubscribe = subscribe( () => {
+					unsubscribe();
+					resolve();
+				} );
+			} );
+		};
+
+		// First call returns undefined
+		expect( select( store ).getSite( slug ) ).toEqual( undefined );
+		await listenForStateUpdate();
```

Comment:
Okay, I'll admit I lost track why we need `listenForStateUpdate` :sweat_smile:

Reply:
The `listenForStateUpdate` is necessary because the resolvers are fired async. If we don't wait for the state to update (or cheat it by waiting for the next cycle) then when we test our values, the resolver won't have completed resolving. I found this part of testing resolvers a bit tricky, and wrote up a little more of what I ran into in a P2: pbxlJb-9T-p2